terraform/development/dynamodb.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_dynamodb_table" "activityhistoryapi_dynamodb_table" {
+    name                  = "ActivityHistory"
+    billing_mode          = "PROVISIONED"
+    read_capacity         = 10
+    write_capacity        = 10
+    hash_key              = "targetId"
+    range_key             = "id"
+
+    attribute {
+        name              = "id"
+        type              = "S"
+    }
+
+    attribute {
+        name              = "targetId"
+        type              = "S"
+    }
+

```

Comment:
You are missing the index definition here (and for the other environments).
You need another attribute for the createdAt property and then the index definition itself. See how the notes api does it...